web/src/ferrostar-core.ts

```diff
@@ -0,0 +1,141 @@
+import { LitElement, html, css, unsafeCSS } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import leafletStyles from "leaflet/dist/leaflet.css?inline";
+import L from "leaflet";
+import markerIconUrl from "../node_modules/leaflet/dist/images/marker-icon.png";
+import markerIconRetinaUrl from "../node_modules/leaflet/dist/images/marker-icon-2x.png";
+import markerShadowUrl from "../node_modules/leaflet/dist/images/marker-shadow.png";
+import init, { NavigationController, RouteAdapter } from "ferrostar";
+
+@customElement("ferrostar-core")
+class FerrostarCore extends LitElement {
+  @property()
+  valhallaEndpointUrl: string = "";
+
+  @property()
+  profile: string = "";
+
+  @property({ attribute: false })
+  httpClient?: Function = fetch;
+
+  // FIXME: type
+  @property({ type: Object })
+  locationProvider!: any;
+
+  // FIXME: type
+  @property({ type: Object })
+  costingOptions!: any;
```

Comment:
Just to make suer I understand, the FIXME is because these should eventually be a stronger type, right?

I appreciate your commitment to perfection, but this can probably be a TODO for now (and if my understanding is correct, we should be able to get this merged as-is) :D